src/main/java/com/ecwid/consul/transport/AbstractHttpTransport.java

```diff
@@ -29,15 +32,15 @@ public abstract class AbstractHttpTransport implements HttpTransport {
 	static final int DEFAULT_READ_TIMEOUT = 1000 * 60 * 10; // 10 min
 
 	@Override
-	public HttpResponse makeGetRequest(HttpRequest request) {
+	public <T> HttpResponse<T> makeGetRequest(HttpRequest request, Function<Reader, T> objConverter) {
```

Comment:
This breaks the backwards compatibility right? There are a few non-SOA projects using this library. https://indeed.sourcegraph.com/search?q=context%3Aglobal+indeed%3Aconsul-api&patternType=standard&groupBy=repo 

But  I guess people don't these methods directly so we are good?

Reply:
yeah, we should be good. It's all internal.